============================= test session starts ==============================
platform linux -- Python 3.8.2, pytest-6.0.1, py-1.9.0, pluggy-0.13.1
rootdir: /home/gabriel/C/UNSW/project, configfile: pytest.ini
plugins: hypothesis-5.24.3
collected 123 items

admin_userpermission_change_test.py .....                                [  4%]
auth_login_test.py ......                                                [  8%]
auth_logout_test.py ..                                                   [ 10%]
auth_register_test.py .........                                          [ 17%]
channel_addowner_test.py ........                                        [ 24%]
channel_details_test.py .....                                            [ 28%]
channel_invite_test.py ......                                            [ 33%]
channel_join_test.py .....                                               [ 37%]
channel_leave_test.py .....                                              [ 41%]
channel_messages_test.py .......                                         [ 47%]
channel_removeowner_test.py ........                                     [ 53%]
channels_create_test.py ....                                             [ 56%]
channels_list_test.py ......                                             [ 61%]
channels_listall_test.py ....                                            [ 65%]
clear_test.py .                                                          [ 65%]
echo_http_test.py E                                                      [ 66%]
echo_test.py ..                                                          [ 68%]
message_edit_test.py ........                                            [ 74%]
message_remove_test.py ........                                          [ 81%]
message_send_test.py .....                                               [ 85%]
search_test.py ...                                                       [ 87%]
user_all_test.py F                                                       [ 88%]
user_profile_setemail_test.py ...                                        [ 91%]
user_profile_sethandle_test.py ....                                      [ 94%]
user_profile_setname_test.py .....                                       [ 98%]
user_profile_test.py ..                                                  [100%]

==================================== ERRORS ====================================
_________________________ ERROR at setup of test_echo __________________________

    @pytest.fixture
    def url():
        url_re = re.compile(r' \* Running on ([^ ]*)')
        server = Popen(["python3", "src/server.py"], stderr=PIPE, stdout=PIPE)
        line = server.stderr.readline()
        local_url = url_re.match(line.decode())
        if local_url:
            yield local_url.group(1)
            # Terminate the server
            server.send_signal(signal.SIGINT)
            waited = 0
            while server.poll() is None and waited < 5:
                sleep(0.1)
                waited += 0.1
            if server.poll() is None:
                server.kill()
        else:
            server.kill()
>           raise Exception("Couldn't get URL from local server")
E           Exception: Couldn't get URL from local server

echo_http_test.py:30: Exception
=================================== FAILURES ===================================
________________________________ test_users_all ________________________________

    def test_users_all():
        other.clear()
    
        users = [
            {
                'fname': 'fname',
                'lname': 'lname',
                'email': 'validemail@gmail.com'
            },
            {
                'fname': 'fname1',
                'lname': 'lname1',
                'email': 'validemail1@gmail.com'
            },
            {
                'fname': 'fname2',
                'lname': 'lname2',
                'email': 'validemail2@gmail.com'
            },
            {
                'fname': 'fname3',
                'lname': 'lname3',
                'email': 'validemail3@gmail.com'
            },
            {
                'fname': 'fname4',
                'lname': 'lname4',
                'email': 'validemail4@gmail.com'
            }
        ]
    
        result = []
    
        #i = 0
        for user in users:
            auth.auth_register(user["email"], "GenericPwd1", user["fname"], user["lname"])
            result.append(auth.auth_login(user["email"], "GenericPwd1"))
            #i += 1
    
        fetched_users = users_all(result[0]['token'])['users']
    
        for i, user in enumerate(fetched_users):
            assert user['email'] == users[i]['email']
            assert user["name_last"] == users[i]['lname']
            assert user['name_first'] == users[i]['fname']
    
    #    auth.auth_register("validemail@gmail.com", "password123", "fname", "lname")
    #    result = auth.auth_login("validemail@gmail.com", "password123")
    
    #    auth.auth_register("validemail1@gmail.com", "password123", "fname1", "lname1")
    #    result1 = auth.auth_login("validemail1@gmail.com", "password123")
    
    #    auth.auth_register("validemail2@gmail.com", "password123", "fname2", "lname2")
    #    result2 = auth.auth_login("validemail2@gmail.com", "password123")
    
    #    auth.auth_register("validemail3@gmail.com", "password123", "fname3", "lname3")
    #    result3 = auth.auth_login("validemail3@gmail.com", "password123")
    
    #    auth.auth_register("validemail4@gmail.com", "password123", "fname4", "lname4")
    #    result4 = auth.auth_login("validemail4@gmail.com", "password123")
    
>       assert users_all(result[0]["token"]) == users
E       AssertionError: assert {'users': [{'...name4', ...}]} == [{'email': 'v...e': 'lname4'}]
E         Use -v to get the full diff

user_all_test.py:85: AssertionError
=========================== short test summary info ============================
FAILED user_all_test.py::test_users_all - AssertionError: assert {'users': [{...
ERROR echo_http_test.py::test_echo - Exception: Couldn't get URL from local s...
==================== 1 failed, 121 passed, 1 error in 0.38s ====================
